### DISCUSSION TITLE
Request for commit history

### DISCUSSION THREAD

**Discussion by ghost** [community/community#24771] [Apps, API and Webhooks]
<p>Hello,<br>
I’m looking to get a list of all commit timestamps and commit messages for my account in the past six months. I’ve looked around the GitHub website but there doesn’t seem to be an option for commit or data export.</p>
<p>Any help would be appreciated. Thanks!</p>

**Comment by ghost** [comment]
<p>Update: I found an export data function in the account tab of my settings, however it doesn’t contain the information I’m looking for. Also, to clarify, I only need information about my commits from repositories that I own. I think it may be possible to find this information by pulling all my repos and doing some scripting with git, but I’m not sure where to start.</p>

**Comment by ghost** [comment]
<p>I figured it out with a bash script. I’ll post it later in case it helps anyone else.</p>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/applemonkey496">@applemonkey496</a> :wave:</p>
<p>If you do feel like sharing your solution and marking your thread as “Solved,” that would be amazing!</p>

**Comment by ghost** [comment, marked as answer]
<p>I put my script in a GitHub repo at <a href="https://github.com/applemonkey496/allgithub" rel="noopener nofollow ugc">applemonkey496/allgithub</a>. I’ve added some basic usage documentation. Hopefully this helps!</p>
